### PR TITLE
feat: refresh dashboard tiles separately

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

stacked on top of #9524 

Tracking the refresh status of insights assumes only one possible cached version of the insight but once #9524 is merged and once insights are being added to multiple dashboards there will be more than one possible cached version

## Changes

Adds new fields to dashboard tiles (that mirror those used to track refresh status of insights) and uses them to track the tile's refreshing status

Insight API reports the last refreshed date of either the insight itself or the insight/dashboard status depending on which is being viewed

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

unit tests and local testing

### CODE REVIEW

File: posthog/api/insight.py [hunk 1]
```diff
@@ -284,13 +300,24 @@ def get_last_refresh(self, insight: Insight):
...
+        dashboard_tile = self.dashboard_tile_from_context(insight, self.context.get("dashboard", None))
+
         result = self.get_result(insight)
         if result is not None:
-            return insight.last_refresh
-        if insight.last_refresh is not None:
-            # Update last_refresh without updating "updated_at" (insight edit date)
-            insight.last_refresh = None
-            insight.save(update_fields=["last_refresh"])
+            if dashboard_tile is not None:
```
Comment (mariusandra) [inline posthog/api/insight.py:307]: Here and elsewhere, there shouldn't be need to write `is not None`

```suggestion
            if dashboard_tile:
```

File: posthog/api/insight.py [hunk 2]
```diff
@@ -162,12 +171,20 @@ class Meta:
             "is_sample",
             "effective_restriction_level",
             "effective_privilege_level",
+            "refreshing",
         )
 
+    @lru_cache(maxsize=1)  # each serializer instance should only deal with one insight/tile combo
+    def dashboard_tile_from_context(self, insight: Insight, dashboard: Optional[Dashboard]) -> Optional[DashboardTile]:
+        dashboard_tile: Optional[DashboardTile] = None
+        if dashboard is not None:
```
Comment (mariusandra) [inline posthog/api/insight.py:180]:
```suggestion
        if dashboard:
```

File: posthog/tasks/update_cache.py [hunk 3]
```diff
@@ -51,19 +51,41 @@
...
+    finally:
+        timer.stop()
+
+    return result
+
 
+def _update_cache_for_queryset(
+    cache_type: CacheType, filter: Filter, key: str, team: Team, queryset: QuerySet
+) -> Optional[List[Dict[str, Any]]]:
+    if queryset.count() == 0:
```
Comment (mariusandra) [inline posthog/tasks/update_cache.py:85]:
```suggestion
    if not queryset.exists():
```

**Review by mariusandra — APPROVED** [review]
Looks good, but there's still this situation with the UX:


Is this in scope for this PR or the next followup?

Some nits/suggestions inline as well.

**Comment by pauldambra** [comment]
> Is this in scope for this PR or the next followup?

Yep, that's in #9604

**Pull request merged**